code/__HELPERS/icons.dm

```diff
@@ -943,6 +943,22 @@ var/global/list/friendly_animal_types = list()
 		return J
 	return 0
 
+/atom/proc/cut_overlays()
+	overlays.Cut()
+	overlays += priority_overlays
+
+/atom/proc/add_overlay(image, priority = 0)
+	if(image in overlays)
+		return
+	var/list/new_overlays = overlays.Copy()
+	if(priority)
+		priority_overlays += image
+		new_overlays += image
+	else
+		new_overlays -= priority_overlays
+		new_overlays += image
+		new_overlays += priority_overlays
+	overlays = new_overlays

```

Comment:
Does this system take into account appearances? If I add an object to new_overlays does doing ``overlays = new_overlays`` use that object's appearance?